qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,322 @@
+        """Transforms a `Property` object.
+
+        This is a recursive reduction, iterating `GroupedProperty` objects when encountering one.
+
+        Args:
+            property: the property object to transform.
+
+        Returns:
+            The transformed property object.
+
+        Raises:
+            TypeError: if an unexpected Property subtype is encountered.
+        """
+        transformed_property: Property
+        # Code for recursion is copied from ActivateSpaceTransformer
+        if isinstance(prop, GroupedProperty):
+            transformed_property = deepcopy(prop)
+
+            for internal_property in iter(prop):
+                try:
+                    transformed_internal_property = self._transform_property(internal_property)
+                    if transformed_internal_property is not None:
+                        transformed_property.add_property(transformed_internal_property)
+                    else:
+                        transformed_property.remove_property(internal_property.name)
+                except TypeError:
+                    logger.warning(
+                        "The Property %s of type %s could not be transformed!",
+                        internal_property.name,
+                        type(internal_property),
+                    )
+                    continue
+
+            if len(transformed_property._properties) == 0:
+                transformed_property = None
+
+        elif isinstance(prop, ElectronicEnergy):
+            h_1 = prop.get_electronic_integral(ElectronicBasis.MO, 1).get_matrix()
+            h_2 = prop.get_electronic_integral(ElectronicBasis.MO, 2).get_matrix()
+
+            hr_1, hr_2 = self._restricted_electronic_integrals(h_1, h_2)
+
+            transformed_property = ElectronicEnergy.from_raw_integrals(
+                ElectronicBasis.MO, hr_1, hr_2
+            )
+            transformed_property._shift = prop._shift.copy()
+            transformed_property.nuclear_repulsion_energy = prop.nuclear_repulsion_energy
+            transformed_property.reference_energy = prop.reference_energy
+
+            transformed_property.second_q_ops = partial(  # type: ignore[assignment]
+                _second_q_ops_electronic_energy, transformed_property
+            )
+
+        elif isinstance(prop, ParticleNumber):
+            transformed_property = prop
+
+            transformed_property.second_q_ops = partial(  # type: ignore[assignment]
+                _second_q_ops_particle_number, transformed_property
+            )
+        elif isinstance(prop, DriverMetadata):
+            transformed_property = prop
+        elif isinstance(prop, SecondQuantizedProperty):
+            transformed_property = None
+        elif isinstance(prop, ElectronicBasisTransform):
+            transformed_property = None
+
+        else:
+            raise TypeError(f"{type(prop)} is an unsupported Property-type for this Transformer!")
```

Comment:
Since some properties are dropped, why not just warn and drop this too rather than make it an error. At least this way things would be a bit more future proof in that if additional properties are added for the problem that this will continue to work - though of course maybe it can be extended to better support such a property rather than dropping it.